update from upstream

jfluecki opened the pull request.

lukasc-ubc commented:
Not sleeping?  Did you get it working?


Lukas Chrostowski
Associate Professor, ECE
University of British Columbia
604-822-8507
http://www.mina.ubc.ca/lukasc
http://siepic.ubc.ca

> On Nov 20, 2015, at 8:22 PM, Jonas Flueckiger <notifications@github.com> wrote:
> 
> Merged #1.
> 
> —
> Reply to this email directly or view it on GitHub.
>

jaspreetj committed:
Merge pull request #47 from lukasc-ubc/master

Update from Master

jaspreetj committed:
Merge pull request #48 from lukasc-ubc/master

update from master

lukasc-ubc committed:
CML gds2fdtd, for KLayout-FDTD integration

rounaksingh committed:
New feature: difference of waveguide length. Working similar to snapping components.

lukasc-ubc committed:
automated coordinate extraction update

format: % X-coord, Y-coord, Polarization, wavelength, type, deviceID,
comment

lukasc-ubc committed:
GSiP Ring mod - added credits

lukasc-ubc committed:
deleting old CML files

lukasc-ubc committed:
Example layout updates, additional simulation data for dc CML

jaspreetj committed:
Merge pull request #49 from lukasc-ubc/master

update from master

jaspreetj committed:
Update SiEPIC_EBeam_functions.lym

fixed netlist trim
Now the script does not include the circuits with no laser attached.

lukasc-ubc committed:
fix for Ports in FDTD for dc_halfring_straight

lukasc-ubc committed:
fix for Ports in FDTD for dc_halfring_straight

lukasc-ubc committed:
Merge pull request #153 from jaspreetj/master

Fixed netlist trim

lukasc-ubc committed:
saving as text gds

lukasc-ubc committed:
Calibre DRC interface to remote server; OSX only version

lukasc-ubc committed:
GSiP Transceiver example with test bench script

mustafacc committed:
Merge remote-tracking branch 'refs/remotes/lukasc-ubc/master'

mustafacc committed:
Added Mustafa PCells library

Includes waveguide arc structures (buggy), early bend structures

mustafacc committed:
Finished bent coupler bus macro

Waveguide arc is still buggy

mustafacc committed:
New Pcell framework and device layers for current Pcells

*Added framework for bent-coupled half ring pcell
-Using this framework crashes kLayout for now, so dont

*Added device layer for bent coupler

mustafacc committed:
Added bent coupled half ring PCell

Completed bent coupled half ring PCell with device recognition layer

mustafacc committed:
Bug fixes

-Fixed Waveguide_Arc PCell issues with device rec and pin rec not
aligning

-Fixed Bent Coupled Half Ring PCell issue with ring's pin rec not
aligning

mustafacc committed:
Bug fixes, PCells minor changes

-Fixed bug where pins of "waveguide arc" PCell would blow out at theta =
0/180 degrees

-Removed "CIRCLE" PCell
-Added framework for "Tapered Ring" PCell

lukasc-ubc committed:
fix for Layout_DoubleBus_RingResonator

lukasc-ubc committed:
focusing sub-wavelength grating coupler PCell - draft

Commit:
Updated README for readability

* centered images
* nested lists
* correct section headings

lukasc-ubc committed:
Merge pull request #155 from nsmith5/master

Updated README for readability

lukasc-ubc committed:
error check for segment function

error happened when a pin had 3 vertices instead of 2

lukasc-ubc committed:
fix for automated measurements

jaspreetj committed:
Merge pull request #50 from lukasc-ubc/master

update master

jaspreetj committed:
FIX  Issue#130 BUG: Waveguide Resize Function

lukasc-ubc committed:
Merge pull request #157 from jaspreetj/master

Fixed issue #130

lukasc-ubc committed:
Merge pull request #152 from rounaksingh/devel

New feature: difference of waveguide length.

lukasc-ubc committed:
new PCell klayout: strip to slot converter

https://www.osapublishing.org/oe/fulltext.cfm?uri=oe-21-16-19029&id=259920

lukasc-ubc committed:
Merge branch 'master' of https://github.com/lukasc-ubc/SiEPIC_EBeam_PDK

mustafacc committed:
Added "Bezier Bend" PCell

Added a new Bezier bend pcell influenced by J. Flueckiger and L.
Chrostowski's bezier bend ample script.

Includes devRec and pinRec layers.

lukasc-ubc committed:
object snapping

improvements with more error checking - specifically check if the objects are instances to address issue #158

lukasc-ubc committed:
waveguide length diff - fixes in error messages

mustafacc committed:
"Bezier Bend' PCell bug fixes

-Fixed bend-straight interface discontinuity bug with "bezier bend"
PCell

mustafacc committed:
Merge remote-tracking branch 'refs/remotes/lukasc-ubc/master'

lukasc-ubc committed:
added layer for Last Metal pad open (layer 13, MLopen)

lukasc-ubc committed:
SEM image list generator

mustafacc committed:
Merged PCells into SiEPIC library

-Removed "Bent_Coupled_Bus" PCell
-Removed "Tapered_Ring" PCell

-Fix EBeam_DC_Seriesrings s-parameters generation script bug

mustafacc committed:
Merge remote-tracking branch 'refs/remotes/lukasc-ubc/master'

mustafacc committed:
Properly merged PCells into SiEPIC library

Merged with the new library of slot-strip and GC PCells

mustafacc committed:
Added "Bent_CDC_Half_Ring" PCell

Bent CDCs half ring for FSR free response rings

mustafacc committed:
Changes and bug fixes to "Bent_CDC_half_ring" PCell

-Added bus bend to straight section with user-selectable bend radius

-Added ability to change ring and bus corrugations seperately

-Fixed bug with disconnected CDC to non-CDC portion in ring

lukasc-ubc committed:
Merge pull request #154 from mustafacc/master

Bent couplers PCells

lukasc-ubc committed:
fix for Bezier_Bend PCell

bezier parameter needed to be a integer - was being rounded to 0.

changed all to TypeDouble, and to microns instead of nm.

lukasc-ubc committed:
cavity hole, example inversion

mustafacc committed:
Bug fixes to "Bent_CDC_Half_Ring" PCell

-Removed discontinuities between bus-bend and bus-straight sections.

-Redefined the gamma variable to N representing number of corrugation sets

-Added pin rec layers to the bus waveguides for auto-snapping

mustafacc committed:
Merged with new cavity hole PCell and bezier bend fixes

mustafacc committed:
Added "Tapered Ring" PCell

Added Tapered ring PCell with user configurable widths at the top and bottom of the ring.

mustafacc committed:
Added "Tapered Ring" PCell, and updates/fixes to Mustafa PCells

-Changes to "Bent CDC Half Ring PCell", user-selectable bus-to-straight section radius, gamma factor replaced by total number of corrugations, fixed non-flush sections

-Added "Tapered Rings"

-Fixes to "Bezier bends" for smoother polygons

lukasc-ubc committed:
Merge branch 'master' into master

lukasc-ubc committed:
Merge pull request #162 from mustafacc/master

Tapered Rings PCell, fixes to "Bent CDC Half Ring", updates to "Bent Coupled Half Ring"

lukasc-ubc committed:
Updated CML, for phase shifted Bragg gratings.

Bragg grating phase #164

lukasc-ubc committed:
fix bad character in EBeam_functions, which didn't work on linux

jaspreetj committed:
Merge pull request #51 from lukasc-ubc/master

update from master

jaspreetj committed:
Update SiEPIC_EBeam_functions.lym

lukasc-ubc committed:
Merge pull request #166 from jaspreetj/master

Fix  error in generate_short_spice_files

lukasc-ubc committed:
Simulations: default to "Trim netlist"

lukasc-ubc committed:
INTERCONNECT: trim netlist by default update

lukasc-ubc committed:
GDS EBeam update: 1310 nm grating couplers, added broadband couplers: from McGill

lukasc-ubc committed:
SEM image list - window

lukasc-ubc committed:
EBeam GDS: added new 1310 GCs and adiabatic components

lukasc-ubc committed:
documentation update

lukasc-ubc committed:
GDS: new TM broadband directional coupler

lukasc-ubc committed:
Remove everything except for the EBeam technology folder

Moving all functionality to a new repo "SiEPIC Tools"

lukasc-ubc committed:
Documentation: Oscar's O-Band GC paper

lukasc-ubc committed:
Update README.md

lukasc-ubc committed:
Update README.md

lukasc-ubc committed:
fix for Trans()

lukasc-ubc committed:
"Waveguide" layer, GSiP

Defined new layer "Waveguide", which points to layer 1/0. Idea: All PDKs can have different layer tables/names, but some need to be consistent. i.e., there is always a default Waveguide layer.

Defined a new layer group "Extra", in which I added Waveguide.

lukasc-ubc committed:
GSiP in tech folder, fixes for 0.25/Brett

Using "get_technology_by_name(GSiP)" rather than currently technology, since the library is loaded when there is layout view!

lukasc-ubc committed:
fixes to EBeam PCells & GDS

in particular, waveguide_bump.

Split EBeam into 2 libs: -dev.

lukasc-ubc committed:
PCM Bragg Spiral, by Stephen

lukasc-ubc committed:
fix for ebeam_crossing pins

lukasc-ubc committed:
EBeam gds, crossing4, pin fix

lukasc-ubc committed:
EBeam Lib: split into Lib and Dev.

lukasc-ubc committed:
pins facing outwards

mustafacc committed:
SiEPIC-Tools compatability fixes

EBeam technology layer names changed to match layer names used in GSiP.

Compatiblity fixes to the following PCells:
*Cavity Holes
*DirectionalCoupler_HalfRing_Arc
*Strip to slot

lukasc-ubc committed:
Layout Invert tone resist

slow. Calibre is much faster.

lukasc-ubc committed:
#32

Rather than translating a path arc, I think we need to go back to a single polygon defined by two arcs. that way the corners are defined precisely.

lukasc-ubc committed:
arc_wg

lukasc-ubc committed:
arc_wg, fixed. DirectionalCoupler_HalfRing_Arc

lukasc-ubc committed:
Merge all the waveguide shapes, to avoid any small gaps,

lukasc-ubc committed:
EBeam v0.3.0 library cleanup

lukasc-ubc committed:
cleanup, technologies

lukasc-ubc committed:
change layer 733_Lumerical to Lumerical

lukasc-ubc committed:
fixes for 0.25

lukasc-ubc committed:
debugging pya.Trans(0,0) 0.25 bug

lukasc-ubc committed:
rename

lukasc-ubc committed:
bragg PCell fix

lukasc-ubc committed:
EBeam CML update

lukasc-ubc committed:
EBeam -dev gds library

Changing Library to EBeam (rather than EBeam_v1.2)
Splitting GDS file into two

mustafacc committed:
PCells fixes for SiEPIC Tools

In SiEPIC_EBeam Library:
Waveguide_Bend

in SiEPIC_Ebeam-dev Library:
Bent_Contra_DC
Bent_Coupled_Half_Ring
Bezier_Bend

lukasc-ubc committed:
Verification

lukasc-ubc committed:
Merge branch 'v0.3.0' of https://github.com/lukasc-ubc/SiEPIC_EBeam_PDK into v0.3.0

mustafacc committed:
PCell fixes for SiEPIC-Tools

Spiral PCell still needs to be fixed

lukasc-ubc committed:
Design Competition

lukasc-ubc committed:
Design for Test DFT.xml file; Verification GDS file

lukasc-ubc committed:
updated layouts

lukasc-ubc committed:
"Component=" text fields for simulation

lukasc-ubc committed:
competition

lukasc-ubc committed:
Merge branch 'v0.3.0'

lukasc-ubc committed:
Update README.md
